spec.html

```diff
@@ -36892,7 +36892,14 @@ <h1>RegExp.prototype [ @@match ] ( _string_ )</h1>
             1. Return ? RegExpExec(_rx_, _S_).
           1. Else,
             1. Assert: _global_ is *true*.
-            1. Let _fullUnicode_ be ! ToBoolean(? Get(_rx_, *"unicode"*)).
+            1. Let _fullUnicode_ be *false*
+            1. Let _hasUnicodeSetsFlag_ be ! ToBoolean(? Get(_rx_, *"unicodeSets"*)).
+            1. If _hasUnicodeSetsFlag_ is *true*, then
+              1. Set _fullUnicode_ to *true*.
+            1. Else,
+              1. Let _hasUnicodeFlag_ be ! ToBoolean(? Get(_rx_, *"unicode"*)).
+              1. If _hasUnicodeFlag_ is *true*, then
+                1. Set _fullUnicode_ to *true*.
```

Comment:
This one is kind of interesting. The naive way to do it would be to (observably) check both getters, and then check if either result was true. However, this phrasing avoids calling the second getter redundantly if the first one already returned `true`. We have to decide which getter is checked first, and is thus being “optimized” for. I decided to optimize for the case where `v` is used, since that’s the flag with the most functionality.

Reply:
Can the spec not express an "or-else" like programming languages do?
I suspect that the performance of the spec pseudocode does not matter much.
I don't have an opinion on whether this should be optimized :-)

Reply:
Yeah I really wanted to just say "or" but I didn't find any precedent for that in other normative algorithms in the spec. Happy to simplify if that’d indeed be okay. Maybe @michaelficarra knows?

I’m less concerned about the “performance“ of this pseudo-code and more about user-observable behavior: it must be clear to engines which getters are/aren’t being accessed, and in which order.

Reply:
Neat! I’ve applied that suggestion here and below. Thanks!